tests/queries/0_stateless/01455_shard_leaf_max_rows_bytes_to_read.sql

```diff
@@ -0,0 +1,20 @@
+DROP TABLE IF EXISTS test_local;
+DROP TABLE IF EXISTS test_distributed;
+
+CREATE TABLE test_local (date Date, value UInt32) ENGINE = MergeTree(date, date, 8192);
+CREATE TABLE test_distributed AS test_local ENGINE = Distributed(test_cluster_two_shards, currentDatabase(), test_local, rand());
+
+INSERT INTO test_local SELECT '2000-08-01', number as value from numbers(50000);
+
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read_leaf = 40000; -- { serverError 158 }
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_bytes_to_read_leaf = 40000; -- { serverError 307 }
+
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read = 60000; -- { serverError 158 }
+SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read_leaf = 60000;
```

Comment:
It is set to `60k` because the following check is affected by measurement error up to 
`max_rows_to_read_leaf  + index_granularity` value, which would be `50000 + 8192`: https://github.com/ClickHouse/ClickHouse/blob/7524a6d412f01ed455dcc36f1ba334f5ba4e332c/src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp#L625